### DISCUSSION TITLE
Set locale programatically

### DISCUSSION THREAD

**Discussion by laurent512** [amannn/next-intl#407] [Q&A]
Hello,

I am using Nextjs 13 with app router.

Is there a way to set the locale programmaticaly i.e. not using Link ? 
There is no information currently available on the https://next-intl-docs.vercel.app/docs/routing/navigation

Thank you in advance !

**Comment by amannn** [comment]
Thank you for the question! I've just added added improved support for this use case, please see https://github.com/amannn/next-intl/issues/408#issuecomment-1637492614
